### Package imports

In [32]:
import pandas as pd
import numpy as np
#import texthero as hero # MUST BE V1.0.5 (pip install texthero==1.0.5)
import seaborn as sns
import matplotlib.pyplot as plt
from collections import Counter

### Reading in the data

In [7]:
train = pd.read_csv("AfrikaansPOSData/train.csv")


### Making sure data looks correct

In [12]:
train.head(5)

Token  POS
0        Die   LB
1       doel  NSE
2        van  SVS
3        die   LB
4  webtuiste  NSE

In [2]:
train.tail(5)

NameError: name 'train' is not defined

Initial checks indicate the presence of missing values. Looking at the dataframe, we see there are NaN rows after each sentence ends.

In [12]:
train['Token'].isna().sum() 

2613

We have 2613 NaN rows in training data. We can also infer that these are many how sentences are contained in the dataset.


In [39]:
train.dropna(inplace=True) # Dropping NaNs

In [ ]:
train.shape

In [44]:
# check total words in vocabulary
wordCounts = Counter(train.Token)
wordCounts = sorted(wordCounts.items(), key=lambda x: x[1], reverse=True)

tagCounts = Counter(train.POS)
tagCounts = sorted(tagCounts.items(), key=lambda x: x[1], reverse=True)

In [43]:
wordCounts[0:10]

[('die', 3872),
 (',', 2086),
 ('van', 1983),
 ('.', 1806),
 ('en', 1634),
 ("'n", 1115),
 ('te', 995),
 ('wat', 936),
 ('in', 922),
 ('is', 909)]

In [46]:
tagCounts

[('SVS', 6931),
 ('NSE', 6126),
 ('LB', 4223),
 ('NSM', 3415),
 ('ASA', 2970),
 ('VTHOG', 2901),
 ('ZM', 2494),
 ('KN', 2326),
 ('ZE', 1847),
 ('BS', 1646),
 ('PB', 1411),
 ('VTUOM', 1254),
 ('LO', 1115),
 ('KO', 1020),
 ('UPI', 990),
 ('VVHOG', 890),
 ('NM', 726),
 ('VTHOK', 719),
 ('VTUOP', 679),
 ('NEE', 565),
 ('RS', 561),
 ('VTHOO', 533),
 ('ASP', 528),
 ('PEMP', 493),
 ('PA', 430),
 ('VUOT', 396),
 ('PDOENP', 355),
 ('UPW', 355),
 ('VVUOP', 353),
 ('ZPL', 340),
 ('PEMB', 338),
 ('ZPR', 331),
 ('UPO', 316),
 ('PO', 291),
 ('THAO', 278),
 ('VTHSG', 266),
 ('UPS', 228),
 ('PTENP', 221),
 ('THAB', 206),
 ('RL', 168),
 ('RWD', 140),
 ('UPV', 134),
 ('VTHSO', 123),
 ('PTEB', 119),
 ('PDMP', 109),
 ('PDMB', 105),
 ('UXD', 88),
 ('VVHOK', 78),
 ('TRAB', 77),
 ('VVHOO', 75),
 ('THBO', 71),
 ('RV', 70),
 ('RK', 67),
 ('PEENP', 67),
 ('PDHEB', 63),
 ('BV', 62),
 ('UPD', 61),
 ('VTUOA', 56),
 ('AOA', 48),
 ('AVA', 43),
 ('VTHOV', 38),
 ('RA', 34),
 ('PDHENP', 26),
 ('THPB', 26),
 ('VVUOM', 2